doc/tracing.md

```diff
@@ -0,0 +1,237 @@
+
+**Installation:**
+```bash
+pip install opentelemetry-sdk opentelemetry-exporter-otlp-proto-http opentelemetry-instrumentation-openai-v2
+```
+
+**Setup:**
+```python
+from opentelemetry import trace
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+from opentelemetry.instrumentation.openai_v2 import OpenAIInstrumentor
+
+# Set up tracer provider
+trace.set_tracer_provider(TracerProvider())
+
+# Configure OTLP exporter
+otlp_exporter = OTLPSpanExporter(
+    endpoint="http://localhost:4318/v1/traces"
+)
+
+# Add span processor
+trace.get_tracer_provider().add_span_processor(
+    BatchSpanProcessor(otlp_exporter)
+)
+
+# Enable OpenAI instrumentation
+OpenAIInstrumentor().instrument()
+```
+</details>
+
+<details>
+<summary>OpenAI - TypeScript/JavaScript</summary>
+
+**Installation:**
+```bash
+npm install @traceloop/instrumentation-openai @traceloop/node-server-sdk
+```
+
+**Setup:**
+```javascript
+const { initialize } = require("@traceloop/node-server-sdk");
+initialize({
+    appName: "opentelemetry-instrumentation-openai-traceloop",
+    baseUrl: "http://localhost:4318",
+    disableBatch: true,
+});
+```
+</details>
+
+## A full example
+
+Here's a complete working example using OpenAI with Python that demonstrates how to set up both the tracing provider and instrumentation:
+
+### Installation
+```bash
+pip install opentelemetry-sdk opentelemetry-exporter-otlp-proto-http opentelemetry-instrumentation-openai-v2
+```
+### Python Code
+
+```python
+import os
+
+### Set up for OpenTelemetry tracing ###
+os.environ["AZURE_TRACING_GEN_AI_CONTENT_RECORDING_ENABLED"] = "true"
+os.environ["AZURE_SDK_TRACING_IMPLEMENTATION"] = "opentelemetry"
+
+from opentelemetry import trace
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+
+resource = Resource(attributes={
+    "service.name": "opentelemetry-instrumentation-azure-ai-inference"
+})
+provider = TracerProvider(resource=resource)
+otlp_exporter = OTLPSpanExporter(
+    endpoint="http://localhost:4318/v1/traces",
+)
+processor = BatchSpanProcessor(otlp_exporter)
+provider.add_span_processor(processor)
+trace.set_tracer_provider(provider)
+
+from azure.ai.inference.tracing import AIInferenceInstrumentor
+AIInferenceInstrumentor().instrument()
+### Set up for OpenTelemetry tracing ###
+
+from azure.ai.inference import ChatCompletionsClient
+from azure.ai.inference.models import UserMessage
+from azure.ai.inference.models import TextContentItem
+from azure.core.credentials import AzureKeyCredential
+
+client = ChatCompletionsClient(
+    endpoint = "https://models.inference.ai.azure.com",
+    credential = AzureKeyCredential(os.environ["GITHUB_TOKEN"]),
```

Comment:
For full example, tell user how to setup this example, for example, we use free GitHub model here, user need to setup the `GITHUB_TOKEN` environment variable. We can add this at the beginning of the code sample (like azure samples)